src/ale.cpp

```diff
@@ -266,4 +254,115 @@ namespace ale {
    return result;
  }
 
+ std::string getPyTraceback() {
+    PyObject* err = PyErr_Occurred();
+    if (err != NULL) {
+        PyObject *ptype, *pvalue, *ptraceback;
+        PyObject *pystr, *module_name, *pyth_module, *pyth_func;
+        char *str;
+        char *full_backtrace;
+        char *error_description;
+
+        PyErr_Fetch(&ptype, &pvalue, &ptraceback);
+        pystr = PyObject_Str(pvalue);
+        str = PyBytes_AS_STRING(PyUnicode_AsUTF8String(pystr));
+        error_description = strdup(str);
+
+        // See if we can get a full traceback
+        module_name = PyUnicode_FromString("traceback");
+        pyth_module = PyImport_Import(module_name);
+        Py_DECREF(module_name);
+
+        if (pyth_module == NULL) {
+            throw runtime_error("getPyTraceback - Failed to import Python traceback Library");
+        }
+
+        pyth_func = PyObject_GetAttrString(pyth_module, "format_exception");
+        PyObject *pyth_val;
+        pyth_val = PyObject_CallFunctionObjArgs(pyth_func, ptype, pvalue, ptraceback, NULL);
+
+        pystr = PyObject_Str(pyth_val);
+        str = PyBytes_AS_STRING(PyUnicode_AsUTF8String(pystr));
+        full_backtrace = strdup(str);
+        Py_DECREF(pyth_val);
+
+        std::string join_cmd = "trace = ''.join(list(" + std::string(full_backtrace) + "))";
+        PyRun_SimpleString(join_cmd.c_str());
```

Comment:
stacktrace is returned as a list, for pretty printing, I am using `PyRun_SimpleString` to join the list as a string. Otherwise you get a `str(list())` type string format.